fastjsonschema/draft04.py

```diff
@@ -349,8 +349,16 @@ def generate_unique_items(self):
         """
         self.create_variable_is_list()
         with self.l('if {variable}_is_list:'):
+            self.l(
+                'def fn(var): '
+                'return frozenset(dict((k, fn(v)) '
+                'for k, v in var.items()).items()) '
+                'if hasattr(var, "items") else tuple(fn(v) '
+                'for v in var) '
+                'if isinstance(var, (dict, list)) else str(var) '
+                'if isinstance(var, bool) else var')
```

Comment:
I'm afraid this is not good solution considering performance. But will affect only when unique items are used and mostly if its dict or list. One of the improvements could be to define this function only once, not with every use. But don't have much time for it, so will accept it and include it in next release for now.

Reply:
Do you have an example of a separate function? Probably I can implement that.

Reply:
There is no feature needing extra logic like this one, so that means adding to generator ability to include this function when needed, and use it here.